cilium-health/launch/endpoint.go

```diff
@@ -102,10 +118,59 @@ func configureHealthRouting(netns, dev string, addressing *models.NodeAddressing
 	return err
 }
 
+// PingEndpoint attempts to make an API ping request to the local cilium-health
+// endpoint, and returns whether this was successful.
+func PingEndpoint() error {
+	if client == nil {
+		return fmt.Errorf("cilium-health endpoint hasn't yet been initialized")
+	}
+	_, err := client.Restapi.GetHello(nil)
+	return err
+}
+
+// CleanupEndpoint attempts to kill any existing cilium-health endpoint and
+// clean up its devices and pidfiles.
+func CleanupEndpoint(owner endpoint.Owner) {
+	pidfile := filepath.Join(option.Config.StateDir, healthPidfile)
+	if pid, err := ioutil.ReadFile(pidfile); err == nil {
+		// Existing cilium-health exists, kill it so we can create a
+		// new one and steal all its logs (om nom nom!)
+		pidInt, err := strconv.Atoi(string(pid))
```

Comment:
This shouldn't ever error but it's a file on-disk so who knows. It might be worth returning early with an error log when `Atoi` fails (or break out of the if).
Separately, perhaps some of this can be moved to the pidfile package? health uses it already, I think.